tests/envs/test_make.py

```diff
@@ -0,0 +1,180 @@
+"""Tests that gym.make works as expected."""
+
+import re
+
+import pytest
+
+import gym
+from gym.envs.classic_control import cartpole
+from gym.wrappers import AutoResetWrapper, OrderEnforcing, TimeLimit
+from gym.wrappers.env_checker import PassiveEnvChecker
+from tests.envs.register_during_make_env import RegisterDuringMakeEnv
+from tests.envs.utils import all_testing_env_specs
+from tests.wrappers.utils import has_wrapper
+
+
+class ArgumentEnv(gym.Env):
+    observation_space = gym.spaces.Box(low=0, high=1, shape=(1,))
+    action_space = gym.spaces.Box(low=0, high=1, shape=(1,))
+
+    def __init__(self, arg1, arg2, arg3):
+        self.arg1 = arg1
+        self.arg2 = arg2
+        self.arg3 = arg3
+
+
+gym.register(
+    id="test.ArgumentEnv-v0",
+    entry_point="tests.envs.test_make:ArgumentEnv",
+    kwargs={
+        "arg1": "arg1",
+        "arg2": "arg2",
+    },
+)
+
+
+def test_make():
+    env = gym.make("CartPole-v1", disable_env_checker=True)
+    assert env.spec.id == "CartPole-v1"
+    assert isinstance(env.unwrapped, cartpole.CartPoleEnv)
+    env.close()
+
+
+def test_make_deprecated():
+    with pytest.raises(
+        gym.error.Error,
+        match=re.escape(
+            "Environment version v0 for `Humanoid` is deprecated. Please use `Humanoid-v4` instead."
+        ),
+    ):
+        gym.make("Humanoid-v0", disable_env_checker=True)
+
+
+def test_make_max_episode_steps():
+    # Default, uses the spec's
+    env = gym.make("CartPole-v1", disable_env_checker=True)
+    assert has_wrapper(env, TimeLimit)
+    assert (
+        env.spec.max_episode_steps == gym.envs.registry["CartPole-v1"].max_episode_steps
+    )
+    env.close()
+
+    # Custom max episode steps
+    env = gym.make("CartPole-v1", max_episode_steps=100, disable_env_checker=True)
+    assert has_wrapper(env, TimeLimit)
+    assert env.spec.max_episode_steps == 100
+    env.close()
+
+    # Env spec has no max episode steps
+    assert gym.envs.registry["test.ArgumentEnv-v0"].max_episode_steps is None
```

Comment:
A more robust way to access the env spec might be `gym.spec`, it's designed for this function basically. Just in case the internals of registration change again.

Reply:
Ahh, I hadn't realised that. I will replace it